backend/src/main/java/site/coduo/pairroomhistory/domain/Timer.java

```diff
@@ -0,0 +1,40 @@
+package site.coduo.pairroomhistory.domain;
+
+import lombok.Getter;
+import site.coduo.pairroomhistory.exception.InvalidTimerException;
+
+@Getter
+public class Timer {
+
+    public static final int FIRST_ROUND = 1;
+
+    private final int timerRound;
+    private final long timerDuration;
+    private final long timerRemainingTime;
+
+    public Timer(final int timerRound, final long timerDuration, final long timerRemainingTime) {
+        validateTime(timerDuration, timerRemainingTime);
+
+        this.timerRound = timerRound;
+        this.timerDuration = timerDuration;
+        this.timerRemainingTime = timerRemainingTime;
+    }
+
+    private void validateTime(long timerDuration, long timerRemainingTime) {
+        if (timerDuration < 0 || timerRemainingTime < 0) {
+            throw new InvalidTimerException("타이머 시간과 남은 시간은 0 이상이어야 합니다.");
+        }
+
+        if (timerDuration < timerRemainingTime) {
+            throw new InvalidTimerException("타이머 시간은 남은 시간보다 작을 수 없습니다.");
+        }
+    }
+
+    public static Timer of(long timerDuration, long timerRemainingTime) {
+        return new Timer(FIRST_ROUND, timerDuration, timerRemainingTime);
+    }
```

Comment:
[질문]
```suggestion

    public Timer(final int timerDuration, final int timerRemainingTime) {
        this(1, timerDuration, timerRemainingTime);
    }
```
해당 정적팩터리를 만든 이유가 따로 있을까용??  또 위와 같이 2개 인자 생성자랑 어떤 차이가 있을까요?

[의견]
개인적으로 정적팩터리 메서드가 어떤 의도가 있는지 파악하기 힘들어요, 가능한 것이라면 생성자로 대체하는 것이 나아 보여요.

Reply:
이 정적팩토리는 타이머 히스토리가 처음 저장됐을 때, 타이머 라운드를 1로 설정하기 위해서 사용했습니다. 기본 생성자와 다른 역할을 부여해주고 싶어서 정팩메를 썼던 것 같아요.
저도 다시 보니 프람이 어떤 부분에서 헷갈려 했는지 이해가 되네요!
정적 팩토리 메서드를 생성자로 변경하였습니다! 좋은 피드백 감사해요👍👍